kod/object/active/holder/nomoveon/battler/player.kod

```diff
@@ -11213,6 +11213,10 @@ messages:
          {
             Debug("ResignFaction miscounted number of faction members!");
          }
+
+         % Prematurely expire the player's active PRINCESS service quest, if present, to avoid faction loss when it organically expires.
```

Comment:
Is the faction loss the core issue, or is it just that it makes sense to cancel the quest if you leave the faction?

Also, the three code blocks for the various factions seem awfully duplicative.  Is it possible to combine them with a loop over the factions, with a few different constants?

Reply:
I think that it's primarily the loss of faction. The penalty "prize" for these quests is faction loss regardless of the faction you started it in and the one you might currently be in. Are you thinking a comment modification is needed?

I'll look into consolidating the code here.

Reply:
Yeah, to me it's more like it doesn't make sense to have a quest from the faction when you've been booted.  Faction loss is a very annoying symptom but not the underlying explanation as to why the quest is being expired.